test/trento/databases/database_test.exs

```diff
@@ -695,6 +760,421 @@ defmodule Trento.Databases.DatabaseTest do
+          health: :passing
+        },
+        [
+          %DatabaseRegistered{
+            database_id: database_id,
+            sid: sid,
+            health: :passing
+          },
+          %DatabaseInstanceRegistered{
+            database_id: database_id,
+            sid: sid,
+            instance_number: instance_number,
+            instance_hostname: instance_hostname,
+            features: features,
+            http_port: http_port,
+            https_port: https_port,
+            start_priority: start_priority,
+            host_id: host_id,
+            system_replication: nil,
+            system_replication_status: nil,
+            health: :passing
+          },
+          %DatabaseTenantsUpdated{
+            database_id: database_id,
+            tenants: tenants
+          }
+        ],
+        %Database{
+          database_id: database_id,
+          sid: sid,
+          tenants: tenants,
+          health: :passing,
+          instances: [
+            %Instance{
+              sid: sid,
+              system_replication: nil,
+              system_replication_status: nil,
+              instance_number: instance_number,
+              features: features,
+              host_id: host_id,
+              health: :passing,
+              absent_at: nil
+            }
+          ]
+        }
+      )
+    end
+
+    test "should emit the DatabaseTenantsUpdated event and update tenants, when the database is already registered but the new registered instance has different tenants" do
+      database_id = Faker.UUID.v4()
+      sid = Faker.StarWars.planet()
+      instance_number = "00"
+      features = Faker.Pokemon.name()
+      host_id = Faker.UUID.v4()
+      tenants = build_list(3, :tenant)
+
+      initial_events = [
+        build(
+          :database_registered_event,
+          database_id: database_id,
+          sid: sid
+        ),
+        build(
+          :database_instance_registered_event,
+          database_id: database_id,
+          sid: sid,
+          instance_number: "10"
+        ),
+        build(:database_tenants_updated_event, database_id: database_id, tenants: tenants)
+      ]
+
+      new_tenants = build_list(2, :tenant)
+
+      assert_events_and_state(
+        initial_events,
+        build(
+          :register_database_instance_command,
+          database_id: database_id,
+          sid: sid,
+          tenants: new_tenants,
+          instance_number: instance_number,
+          features: features,
+          host_id: host_id,
+          health: :passing
+        ),
+        [
+          build(
+            :database_instance_registered_event,
+            database_id: database_id,
+            sid: sid,
+            instance_number: instance_number,
+            features: features,
+            host_id: host_id,
+            health: :passing
+          ),
+          build(:database_tenants_updated_event, database_id: database_id, tenants: new_tenants)
```

Comment:
I find it strange to use a factory in the assert itself, but it works the same, that's true.
In fact, you use factories in some, and real events in others hehe

Reply:
yeah, it's the same compatibility issue as before for minimal changes xD